STM32/AC/Core/Src/ACBoard.c

```diff
@@ -188,63 +224,113 @@ static void CAallOn(bool isOn)
+** the power on all ports by reducing the PWM
+*/
 static void heatSinkLoop()
 {
-	static unsigned long previous = 0;
+    static unsigned long previous = 0;
     // Turn on fan if temp > 55 and turn of when temp < 50.
-    if (heatSinkTemperature < 50 && !isFanForceOn)
+    if (heatSinkTemperature <= MAX_TEMPERATURE)
     {
-		stmSetGpio(fanCtrl, false);
+        if (heatSinkTemperature < 50 && !isFanForceOn)
+        {
+            stmSetGpio(fanCtrl, false);
+        }
+        else if (heatSinkTemperature > 55)
+        {
+            stmSetGpio(fanCtrl, true);
+        }
+
+        bsClearField(BS_OVER_TEMPERATURE_Msk);
     }
-    else if (heatSinkTemperature > 55)
+    else // Safety mode to avoid overheating of the board
     {
-    	stmSetGpio(fanCtrl, true);
+        stmSetGpio(fanCtrl, true);
+        
+        unsigned long now = HAL_GetTick();
+        if (now - previous > 2000)  // 2000ms. Should be aligned with the control period of heater.
+        {
+            previous = now;
+            adjustPWMDown();
+        }
+
+        bsSetError(BS_OVER_TEMPERATURE_Msk);
     }
-    else if (heatSinkTemperature > MAX_TEMPERATURE) // Safety mode to avoid overheating of the board
+
+    setBoardTemp(heatSinkTemperature);
+}
+
+/*!
+** @brief Updates the global error/status object.
+**
+** Adds the port and fan status bits into the error field, and clears the error bit if there are no
+** more errors.
+*/
+static void updateErrorField() 
+{
+    stmGetGpio(fanCtrl) ? bsSetField(AC_BOARD_PORT_x_STATUS_Msk(0)) : bsClearField(AC_BOARD_PORT_x_STATUS_Msk(0));
+    for(int i = 0; i < AC_BOARD_NUM_PORTS; i++)
+    {
+        stmGetGpio(heaterPorts[i].heater) ? bsSetField(AC_BOARD_PORT_x_STATUS_Msk(i+1)) : bsClearField(AC_BOARD_PORT_x_STATUS_Msk(i+1));
+    }
+
+    /* Clear the error mask if there are no error bits set any more. This logic could be done when
+    ** the (other) error bits are cleared, but doing here means it only needs to be done once */
+    if (!(bsGetStatus() & AC_BOARD_No_Error_Msk))
     {
-    	unsigned long now = HAL_GetTick();
-    	if (now - previous > 2000)	// 2000ms. Should be aligned with the control period of heater.
-    	{
-    		previous = now;
-    		adjustPWMDown();
-    	}
+        bsClearField(BS_ERROR_Msk);
     }
 }
 
+/***************************************************************************************************
+** PUBLIC FUNCTIONS
+***************************************************************************************************/
+
 void ACBoardInit(ADC_HandleTypeDef* hadc, WWDG_HandleTypeDef* hwwdg)
 {
-	// Always allow for DFU also if programmed on non-matching board or PCB version.
-	initCAProtocol(&caProto, usb_cdc_rx);
+    // Always allow for DFU also if programmed on non-matching board or PCB version.
+    initCAProtocol(&caProto, usb_cdc_rx);
 
     BoardType board;
     if (getBoardInfo(&board, NULL) || board != AC_Board)
+    {
+        bsSetError(AC_BOARD_VERSION_ERROR);
```

Comment:
Going into this condition (and not executing the rest of the init function) means that the ADC is never initialised. Hence, the ADCMonitorLoop never triggers the printCurrentArray callback function and the status is therefore never printed.

Reply:
OK, sorted